api/v1beta1/cluster_types.go

```diff
@@ -943,11 +956,13 @@ type ClusterStatus struct {
 	// Deprecated: This field is deprecated and is going to be removed in the next apiVersion. Please see https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.md for more details.
 	//
 	// +optional
+	// +kubebuilder:validation:MaxLength=10240
 	FailureMessage *string `json:"failureMessage,omitempty"`
 
 	// phase represents the current phase of cluster actuation.
 	// E.g. Pending, Running, Terminating, Failed etc.
 	// +optional
+	// +kubebuilder:validation:MaxLength=50
 	Phase string `json:"phase,omitempty"`
```

Comment:
Phase is not an enum?

Reply:
Apparently :) I don't know why

Reply:
Made a note on the umbrella issue for v1beta2: https://github.com/kubernetes-sigs/cluster-api/issues/10852

Reply:
Now got your point. Will make all Phase field enums